File: string/basic/check-string-pangrammatic-lipogram/pangrammatic-lipogram.js
const pangrammaticLipogram = s => {
  let alphabets = "abcdefghijklmnopqrstuvwxyz";
  let letterSet = new Set([...alphabets]);
  let string = s.toLowerCase();
  for (let i = 0; i < string.length; i++) {
    letterSet.delete(string[i]);
    if (!letterSet.size) {
      return "Pangram";
    }
  }

  return letterSet.size === 1
    ? "Pangrammatic Lipogram"
    : "Not a Pangram but might a Lipogram";
};

console.log(pangrammaticLipogram("welcome to geeksforgeeks"));

console.log(pangrammaticLipogram("The quick brown fox jump over the lazy dog"));

console.log(
  pangrammaticLipogram("The quick brown fox jumps over the lazy dog")
);

File: string/basic/check-string-pangrammatic-lipogram/pangrammatic-lipogram.py
def pangrammatic_lipogram(s):
    alphabets = "abcdefghijklmnopqrstuvwxyz"
    letter_set = set(list(alphabets))
    s = s.lower()

    for char in s:
        letter_set.discard(char)
        if len(letter_set) == 0:
            return 'Pangram'

    if len(letter_set) == 1:
        return 'Pangrammatic Lipogram'

    return 'Not a Pangram but might a Lipogram'


print(pangrammatic_lipogram("welcome to geeksforgeeks"))

print(pangrammatic_lipogram("The quick brown fox jump over the lazy dog"))

print(pangrammatic_lipogram("The quick brown fox jumps over the lazy dog"))
